### PR TITLE
feat(cli): `--builder.disable` arg

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Add `--builder.disable` option which disables built-in payload builder.

### CODE REVIEW

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -586,9 +586,10 @@ where
                     debug!(target: "consensus::engine", fcu_head_num=?outcome.header().number, current_head_num=?self.blockchain.canonical_tip().number, "Ignoring beacon update to old head");
                 }
 
-                if let Some(attrs) = attrs {
+                if let Some((builder, attrs)) = self.payload_builder.as_ref().zip(attrs) {
```
Comment (mattsse) [inline crates/consensus/beacon/src/engine/mod.rs:589]: is this okay with engine api?
that payload building is optional?

**Review by mattsse — COMMENTED** [review]
what's the upside of having this?

File: crates/consensus/beacon/src/engine/mod.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/consensus/beacon/src/engine/mod.rs:589]: The payload building is not optional, hence this behavior is not the default one.

However, the ability to disable _default_ payload building is beneficial for custom integrations.

**Comment by onbjerg** [comment]
Not familiar with how this is glued together, but I assume that rn `PayloadStore` is the payload builder? they should be separate, no? What is the flow for validators that use flashbots or other means of constructing payloads? I assume they still post those payloads to the execution engine to validate, and in that case we should still store those payloads somewhere. If this is the case, then the payload store should just store payloads we've validated, and the payload builder is some other sort of construction

I might be ignorant on the flashbots/CL/EL flow though

File: crates/consensus/beacon/src/engine/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/consensus/beacon/src/engine/mod.rs:589]: > However, the ability to disable default payload building is beneficial for custom integrations.

what would be a custom integration?

**Pull request closed without merging**